### PR TITLE
CRUD with current user and log

### PR DESCRIPTION
**Pull request by MesropMkhitaryan**

### CODE REVIEW

File: black-mulberry-core/src/main/java/com/black/mulberry/core/mapper/ProductMapper.java [hunk 1]
```diff
@@ -26,4 +27,6 @@ public ProductResponse toResponse(Product product) {
         return modelMapper.map(product, ProductResponse.class);
     }
 
+
+
```
Comment (Mushegh128) [inline black-mulberry-core/src/main/java/com/black/mulberry/core/mapper/ProductMapper.java]: this is unnessacery lines, delete them

File: black-mulberry-core/src/main/java/com/black/mulberry/core/mapper/ProductMapper.java [hunk 2]
```diff
@@ -9,6 +9,7 @@
 import org.springframework.stereotype.Service;
 
 import java.lang.reflect.Type;
+import java.util.List;
```
Comment (Mushegh128) [inline black-mulberry-core/src/main/java/com/black/mulberry/core/mapper/ProductMapper.java]: this is unused import. delete them

File: black-mulberry-core/src/main/java/com/black/mulberry/core/service/ProductService.java [hunk 3]
```diff
@@ -8,15 +10,15 @@
...
-    ProductResponse update(long id, ProductRequest productRequest);
+    Product update(long id, ProductRequest productRequest);
 
     void deleteById(long id);
 
-    ProductResponse findById(long id);
+    Product findById(long id);
 
-    List<ProductResponse> findAll();
+    List<Product> findAll();
```
Comment (Mushegh128) [inline black-mulberry-core/src/main/java/com/black/mulberry/core/service/ProductService.java]: menq paymanavorvel einq or entity chenq return ene, lutshe het poxes

File: black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java [hunk 4]
```diff
@@ -39,57 +41,69 @@ public class ProductServiceImpl implements ProductService {
...
-        Optional<User> optionalUser = userRepository.findById(productRequest.getUserId());
+    public Product save(ProductRequest productRequest,User user) {
+        log.info("request from user: {} to create product",user.getName());
+        Optional<User> optionalUser = userRepository.findById(user.getId());
         Optional<CategoryProduct> categoryProduct = categoryProductRepository.findById(productRequest.getCategoryProductId());
-        if(optionalUser.isEmpty() || categoryProduct.isEmpty()){
-           throw new NullPointerException("user or category is null");
+        if(categoryProduct.isEmpty()){
+            log.debug("category can't be null");
+            throw new NullPointerException("category is null");
```
Comment (Mushegh128) [inline black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java]: @nchi null pointer kqces? sik infakti bune brat, veranaye sax motodt ira logerov handerdz, dir erkar mtace u pordze du qezi sxal ban gtnis u uxxes, inchqan krnas sax kataryali ber or sorvis.

**Review by Mushegh128 — CHANGES_REQUESTED** [review]

File: black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/CategoryProductImpl.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+@RequiredArgsConstructor
+@Slf4j
+public class CategoryProductImpl implements CategoryProductService {
+
+    private final CategoryProductRepository categoryProductRepository;
+
+    @Override
+    public CategoryProduct findById(long categoryId) {
+        CategoryProduct categoryProduct = categoryProductRepository.findById(categoryId).orElseThrow(() -> {
+            log.debug("category with id: {} not found", categoryId);
```
Comment (Mushegh128) [inline black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/CategoryProductImpl.java]: sik DEBUG log che, ari poxenq kam ERROR kak WARN

File: black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java [hunk 6]
```diff
@@ -26,84 +25,105 @@
...
-    public ProductResponse save(ProductRequest productRequest) {
-        Optional<User> optionalUser = userRepository.findById(productRequest.getUserId());
-        Optional<CategoryProduct> categoryProduct = categoryProductRepository.findById(productRequest.getCategoryProductId());
-        if(optionalUser.isEmpty() || categoryProduct.isEmpty()){
-           throw new NullPointerException("user or category is null");
-        }
+    public Product save(ProductRequest productRequest, User user) {
+        log.info("request from user: {} to create product", user.getEmail());
+        // todo after userService changes change to userService
+        Optional<User> userOptional = userRepository.findById(user.getId());
```
Comment (Mushegh128) [inline black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java:44]: servicid mej uninq arden ger user by ID, krnas @dik ogtagorces es nuyn ban@ stexe chexni.

File: black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java [hunk 7]
```diff
@@ -26,84 +25,105 @@
...
+        CategoryProduct categoryById = categoryProductService.findById(productRequest.getCategoryProductId());
         Product product = productMapper.toEntity(productRequest);
-        product.setCategoryProduct(categoryProduct.get());
-        product.setUser(optionalUser.get());
+        product.setCategoryProduct(categoryById);
+        product.setUser(userOptional.get());
```
Comment (Mushegh128) [inline black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java:48]: stex null_i het krna arnchvinq

File: black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java [hunk 8]
```diff
@@ -26,84 +25,105 @@
...
-        return productMapper.toResponse(save);
+        log.info("product with id: {} succesfully created", save.getId());
+        return save;
+    }
+
+    @Override
+    public Product findById(long id) {
+        log.info("Request to get product with id: {}", id);
+        Product product = productRepository.findByIdAndIsDeleteFalse(id).orElseThrow(() -> {
+            log.debug("product with id: {} not found", id);
```
Comment (Mushegh128) [inline black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java]: ERROR kam WARN

File: black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java [hunk 9]
```diff
@@ -26,84 +25,105 @@
...
     @Override
-    public ProductResponse findById(long id){
-        Product product = productRepository.findByIdAndIsDeleteFalse(id).orElseThrow(() -> new ProductNotExistException(
-                "product with id: " + id + " does not exist"
-        ));
-        return productMapper.toResponse(product);
+    public Product findByIdAndUserId(long productId, long userId) {
+        log.info("request to get product with id: {}", productId);
+        Product product = productRepository.findByIdAndIsDeleteFalseAndUserId(productId, userId).orElseThrow(() -> {
+            log.debug("product with id: {} not found", productId);
```
Comment (Mushegh128) [inline black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java]: sik el

File: black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java [hunk 10]
```diff
@@ -26,84 +25,105 @@
...
-    public byte[] getImage(@RequestParam("fileName") String fileName)  {
-        InputStream inputStream = null;
+    public byte[] getImage(String fileName) {
+        log.info("request get image");
+        InputStream inputStream;
         try {
             inputStream = new FileInputStream(folderPath + File.separator + fileName);
-            byte[] bytes = IOUtils.toByteArray(inputStream);
-            return  bytes;
+            return IOUtils.toByteArray(inputStream);
```
Comment (Mushegh128) [inline black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java]: bac connection kmna , sra masin xosacel enq - dir try with resourcesi mej

File: black-mulberry-mvc/src/main/java/com/black/mulberry/mvc/controller/ProductController.java [hunk 11]
```diff
@@ -27,7 +27,7 @@ public class ProductController {
 
     private final CategoryProductRepository categoryProductRepository;
 
-    @Value("${blackMulberry.mvc.images.product}")
+    @Value("${blackMulberry.product.images}")
     private String folderPath;
 
     @GetMapping("/product/add")
```
Comment (Mushegh128) [inline black-mulberry-mvc/src/main/java/com/black/mulberry/mvc/controller/ProductController.java:33]: controlleri vra areq misht ogtagorcenq @RequestMapping("/v1/product") , isk @GetMapping("/product/add")  -> @GetMapping("/add"). hame URL_in nayelov es khasknam or sik praduct avelcnelu url e, bayc metodt uxxaki ej@ kuda , ed pah@ myam veranayeq u hisheq or @GetMaping@ body chuni.

es paragayum krnas nayes tarberak urdex or ej bacelu gorcoxutyun enox URL_nerd kexnin orinak _> "nav/product_new"

File: black-mulberry-rest/src/main/java/com/black/mulberry/rest/endpoint/ProductEndpoint.java [hunk 12]
```diff
@@ -13,35 +17,40 @@
 @RequiredArgsConstructor
 @RequestMapping("/products")
```
Comment (Mushegh128) [inline black-mulberry-rest/src/main/java/com/black/mulberry/rest/endpoint/ProductEndpoint.java]: "v1/products"

**Review by Mushegh128 — CHANGES_REQUESTED** [review]

File: black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java (hunk 6 above)
Comment (hovnaz) [reply black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java:44]: verevy todo e grac, et masy mtacel enk. userService-i mejinery sagh DTO kveradardzne et patcharov cheink krna miangamic service-ic kancheink

File: black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java (hunk 7 above)
Comment (hovnaz) [reply black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java:48]: chisht es bayc menk hashvi enk arel es pahy, chuzecank null-i stugum anteghi grel vordev petq e UserSerive-ic kanchenk hetagayum u UserSerive-i mej et pahy arden hashvi e arac

File: black-mulberry-mvc/src/main/java/com/black/mulberry/mvc/controller/ProductController.java (hunk 11 above)
Comment (hovnaz) [reply black-mulberry-mvc/src/main/java/com/black/mulberry/mvc/controller/ProductController.java:33]: sik imijayloc mvc-n e u sagh petq e poxvi enqan or dzevapoxel enk or build@ ancni

File: black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java (hunk 7 above)
Comment (Mushegh128) [reply black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java:48]: karevor@ chmoranaq

File: black-mulberry-mvc/src/main/java/com/black/mulberry/mvc/controller/ProductController.java (hunk 11 above)
Comment (Mushegh128) [reply black-mulberry-mvc/src/main/java/com/black/mulberry/mvc/controller/ProductController.java:33]: ed url@ shat texer kogtagorcvi? vsyo taki @RequestMaping avelacnel@ url chi poxe motd, bayc v1 avelacnel@ stex cankali e

File: black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java (hunk 7 above)
Comment (hovnaz) [reply black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java:48]: ytor hamar todo enk grel:D

**Review by Mushegh128 — APPROVED** [review]

**Pull request merged**